lib/httpoison/base.ex

```diff
@@ -432,6 +433,29 @@ defmodule HTTPoison.Base do
      end
   end
 
+  defp do_request(method, request_url, request_headers, {:stream, enumerable}, hn_options) do
+    with {:ok, ref} <- :hackney.request(method, request_url, request_headers, :stream, hn_options) do
+
+      failures = Stream.transform(enumerable, :ok, fn
+        _, :error -> {:halt, :error}
+        bin, :ok  -> {[], :hackney.send_body(ref, bin)}
+        _, error  -> {[error], :error}
+      end) |> Enum.into([])
```

Comment:
so if I understood well we always "materialise" the stream to become a list before sending? Then we are not going to "stream" the upload as it will be in memory before sending the request.